src/config.rs

```diff
@@ -353,6 +347,34 @@ impl GitHub {
     }
 }
 
+/// Take a quoted string from the config file and unescape it (i.e. strip the start and end quote
+/// (") characters and process any escape characters in the string.)
+fn unescape_str(us: &str) -> String {
+    // The regex in config.l should have guaranteed that strings start and finish with a
+    // single quote (") character.
+    debug_assert!(us.starts_with('"') && us.ends_with('"'));
+    let mut s = String::new();
+    // We iterate over all characters except the opening and closing quote characters.
+    let mut i = '"'.len_utf8();
```

Comment:
Shall we cache the length of a double quote in a local variable, as we compute it lots of times? I'd also be OK with asserting the length is 1 and using the constant 1 henceforth.

Reply:
They'll almost certainly be turned into a constant by the compiler, and it makes sure we can't forget that characters aren't always 1 byte wide.

Reply:
That rebuttals the second suggestion.

So how about replacing all of the instances of `'"'.len_utf8()` (which is qite hard to read due to syntactical noise) with `dquo_len` where `dquo_len = '"'.len_utf8()`?

Reply:
Both ways suck, but I find the overhead of a variable even higher than the repeated function call.

Reply:
Personally I prefer to have the easily-read variable name. I'd have thought the Rust compiler would inline it to a constant anyway.

I won't insist. Do whatever you think is best and we can move on.

Reply:
When I said "I find the overhead higher" I meant the "intellectual overhead" (the compiler will optimise both cases away).